detectors/node/opentelemetry-resource-detector-aws/src/detectors/AwsEcsDetector.ts

```diff
@@ -38,27 +41,65 @@ import { getEnv } from '@opentelemetry/core';
 export class AwsEcsDetector implements Detector {
   readonly CONTAINER_ID_LENGTH = 64;
   readonly DEFAULT_CGROUP_PATH = '/proc/self/cgroup';
+
   private static readFileAsync = util.promisify(fs.readFile);
 
+  private static async getUrlAsJson(url: string): Promise<any> {
+    return new Promise<string>((resolve, reject) => {
+      const request = http.get(url, (response: any) => {
```

Comment:
There should be a type for the response in the `http` module

Reply:
Done in ab0ad4f5